pkgs/development/python-modules/ctap-keyring-device/default.nix

```diff
@@ -5,28 +5,35 @@
   pythonRelaxDepsHook,
   setuptools-scm,
   # install requirements
+  six,
   fido2,
   keyring,
   cryptography,
   # test requirements
   pytestCheckHook,
+  unittestCheckHook,
+  mock,
 }:
 
 let
   fido2_0 = fido2.overridePythonAttrs (oldAttrs: rec {
     version = "0.9.3";
-    format = "setuptools";
     src = fetchPypi {
       inherit (oldAttrs) pname;
       inherit version;
       hash = "sha256-tF6JphCc/Lfxu1E3dqotZAjpXEgi+DolORi5RAg0Zuw=";
     };
+    build-system = [ setuptools-scm ];
+    propagatedBuildInputs = [ six ];
+    nativeCheckInputs = [ unittestCheckHook ];
+    checkInputs = [ mock ];
   });
 in
 buildPythonPackage rec {
   pname = "ctap-keyring-device";
   version = "1.0.6";
-  format = "setuptools";
+  pyproject = true;
+  build-system = [ setuptools-scm ];
```

Comment:
This should be between postPatch and nativeBuildInputs. It is basically another list of inputs.